src/net/server/mod.rs

```diff
@@ -0,0 +1,245 @@
+use std::net::SocketAddr;
+
+use axum::{
+    extract::rejection::QueryRejection,
+    response::{IntoResponse, Response},
+    routing::get,
+    Router,
+};
+use axum_server::{tls_rustls::RustlsConfig, Handle};
+use hyper::StatusCode;
+use thiserror::Error;
+use tokio::task::JoinHandle;
+
+mod handlers;
+
+#[derive(Error, Debug)]
+pub enum MpcServerError {
+    #[error(transparent)]
+    BadQueryString(#[from] QueryRejection),
+}
+
+impl IntoResponse for MpcServerError {
+    fn into_response(self) -> Response {
+        let status_code = match &self {
+            MpcServerError::BadQueryString(_) => StatusCode::BAD_REQUEST,
+        };
+
+        (status_code, self.to_string()).into_response()
```

Comment:
curious if there's a "standard" way of converting between types in rust. should you prefer `into*` methods or `from*` methods? for instance, in this case does a `Response::from` exist? if so, why use `into_response()` over that?

Reply:
I suspect that the reason for this custom trait to exist is to be able to do dynamic dispatch inside Axum as `From` trait is not object-safe.

in my experience `.into` is the preferred way to convert, but it often not possible because `into` method is not generic, so you fall back to `T::from(..)`